flink-formats/flink-parquet/pom.xml

```diff
@@ -45,6 +45,12 @@ under the License.
 			<scope>provided</scope>
 		</dependency>
 
+		<dependency>
+			<groupId>org.apache.flink</groupId>
```

Comment:
I was a bit worried that all users that use parquet now also get the Avro dependencies, but as far as I can see they already have all avro dependencies % flink-avro (128 kb) already. Can you double-check if this is true?

Reply:
new transitive deps of flink-parquet: 
_+- org.apache.flink:flink-avro:jar:1.12.1:compile
[INFO] |  \- org.apache.avro:avro:jar:1.10.0:compile
[INFO] |     +- com.fasterxml.jackson.core:jackson-core:jar:2.10.1:compile
[INFO] |     \- com.fasterxml.jackson.core:jackson-databind:jar:2.10.1:compile
[INFO] |        \- com.fasterxml.jackson.core:jackson-annotations:jar:2.10.1:compile_

these avro and jackson deps were already transitive deps of hadoop-common
_
**+- org.apache.hadoop:hadoop-common:jar:2.4.1:provided**
[INFO] |  +- org.apache.hadoop:hadoop-annotations:jar:2.4.1:provided
[INFO] |  +- com.google.guava:guava:jar:11.0.2:compile
[INFO] |  +- commons-cli:commons-cli:jar:1.3.1:provided
[INFO] |  +- org.apache.commons:commons-math3:jar:3.5:provided
[INFO] |  +- xmlenc:xmlenc:jar:0.52:provided
[INFO] |  +- commons-httpclient:commons-httpclient:jar:3.1:provided
[INFO] |  +- commons-codec:commons-codec:jar:1.13:provided
[INFO] |  +- commons-io:commons-io:jar:2.7:provided
[INFO] |  +- commons-net:commons-net:jar:3.1:provided
[INFO] |  +- javax.servlet:servlet-api:jar:2.5:compile
[INFO] |  +- org.mortbay.jetty:jetty:jar:6.1.26:provided
[INFO] |  +- org.mortbay.jetty:jetty-util:jar:6.1.26:provided
[INFO] |  +- com.sun.jersey:jersey-core:jar:1.9:provided
[INFO] |  +- com.sun.jersey:jersey-json:jar:1.9:provided
[INFO] |  |  +- org.codehaus.jettison:jettison:jar:1.1:provided
[INFO] |  |  +- com.sun.xml.bind:jaxb-impl:jar:2.2.3-1:provided
[INFO] |  |  +- org.codehaus.jackson:jackson-jaxrs:jar:1.8.3:provided
[INFO] |  |  \- org.codehaus.jackson:jackson-xc:jar:1.8.3:provided
[INFO] |  +- com.sun.jersey:jersey-server:jar:1.9:provided
[INFO] |  |  \- asm:asm:jar:3.1:provided
[INFO] |  +- tomcat:jasper-compiler:jar:5.5.23:provided
[INFO] |  +- tomcat:jasper-runtime:jar:5.5.23:provided
[INFO] |  +- javax.servlet.jsp:jsp-api:jar:2.1:provided
[INFO] |  +- commons-el:commons-el:jar:1.0:provided
[INFO] |  +- commons-logging:commons-logging:jar:1.1.3:provided
[INFO] |  +- net.java.dev.jets3t:jets3t:jar:0.9.0:provided
[INFO] |  |  +- org.apache.httpcomponents:httpclient:jar:4.5.3:compile
[INFO] |  |  +- org.apache.httpcomponents:httpcore:jar:4.4.6:compile
[INFO] |  |  \- com.jamesmurty.utils:java-xmlbuilder:jar:0.4:provided
[INFO] |  +- commons-lang:commons-lang:jar:2.6:compile
[INFO] |  +- commons-configuration:commons-configuration:jar:1.7:provided
[INFO] |  |  +- commons-digester:commons-digester:jar:1.8.1:provided
[INFO] |  |  \- commons-beanutils:commons-beanutils:jar:1.8.3:provided
[INFO] |  +- org.codehaus.jackson:jackson-core-asl:jar:1.8.8:provided
[INFO] |  +- org.codehaus.jackson:jackson-mapper-asl:jar:1.8.8:provided
**[INFO] |  +- org.apache.avro:avro:jar:1.10.0:compile
[INFO] |  |  +- com.fasterxml.jackson.core:jackson-core:jar:2.10.1:compile
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-databind:jar:2.10.1:compile
[INFO] |  |     \- com.fasterxml.jackson.core:jackson-annotations:jar:2.10.1:compile**
_

Reply:
> > @AHeise ParquetInputFormat base class was removed since I submitted my PR hence the compilation issues, commit [ce3631a](https://github.com/apache/flink/commit/ce3631af7313855f675e29b8faa386f6e5a2d43c) removed it. This commit mentions "Use the filesystem connector with a Parquet format as a replacement". I guess it refers to https://ci.apache.org/projects/flink/flink-docs-master/docs/connectors/table/filesystem/ which is SQL based. But what if our pipeline pipeline does not use SQL but DataSet API ?
> 
> It's a good and tough question. I spoke to @twalthr offline and for now the plan is as follows:
> 
> * Table API drops old planner and thus, most of the removed code in that commit is dead code.
> * Table API will support everything running on DataStream that used to work when it ran on DataSet.
> * 1.13 will be the last release with full DataSet support (BatchTableEnvironment will be dropped)
> 
> Since a few features of DataSet are still not supported in DataStream, we expect users to stick to 1.13 for a longer time and probably skip 1.14. So, we'd suggest to merge your 2 PRs to release-1.13 instead of master. Then, all DataSet users would benefit from your contributions while we unblock future developments that would break DataSet as of 1.13.
> 
> If it turns out that the community wants to have these features in 1.14 for some reasons (Table API not as far as it should), we can still re-add `ParquetInputFormat` and forward port your PR before feature freeze in 3 months.
> 
> Note 1: It might still be possible to have a Flink 1.14 DataSet application using 1.13 formats. In general, it's always possible to copy the old code into your own project.
> Note 2: If you are missing combineable aggregations in DataStream, maybe it would be better to move to Table API to begin with. At this point, no-one really knows how much of DataSet will be ported to DataStream. It doesn't really make sense to have 2 APIs with high-level primitives like joins. The main idea is to use Table API by default with a rich user experience and go down to DataStream only when needed (timer, user state, ...).


Hi @AHeise, thanks for your detailed comment.
Merging my 2 PRs to 1.13 looks fine to me, I think backporting to 1.12 might me considered as well as Amazon EMR is still 1.12 based (latest is 1.12.1) and they are likely to move to latest 1.12 in short term releases of EMR.

For the context, what I'm doing is a benchmark comparing Flink Dataset/Datstream/SQL and other technologies such as Apache Beam. So I'd like to have a working DataSet/DataStream pipeline in addition to my existing working SQL one (using blink planner based on DataStream and BatchTableEnvironment). And I guess there will be some users that will want to stick to DataSet/DataStream if they have existing pipelines that they do not want to re-code but just migrate to next flink version.

Reply:
Okay doesn't look too bad. Databind is quite hefty. Maybe we should add this dependency as optional in the same way `parquet-avro` is already optional.

Reply:
fair enough

Reply:
I put flink-avro optional and excluded jackson-databind

Reply:
My bad  jackson-databind is needed for the other tests in the parquet module